### PR TITLE
Remove the "lockbox" custom megazord [ci full].

### PR DESCRIPTION
**Pull request by rfk**
The lockwise apps are on a very old version of appservices and I don't believe
we have a plan for updating them anytime soon. Continuing to build the lockbox
megazord does not seem like a good use of CI or local dev resources.

We understand how to bring this back if we need it, but also, if we do need
it again in future it's probably a good opportunity to revisit how we build
custom megazords in the first place.

### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - `automation/all_tests.sh` runs to completion and produces no failures
  - Note: For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
- ~~[ ] **Tests**: This PR includes thorough tests or an explanation of why it does not~~
  - Not adding anything, so not adding any tests.
- [x] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - ~~Any breaking changes to Swift or Kotlin binding APIs are noted explicitly~~
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - ~~Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.~~

### CODE REVIEW

File: docs/design/megazords.md [hunk 1]
```diff
@@ -52,10 +52,11 @@ to replace the "full-megazord" AAR with a custom-built megazord AAR containing o
 Such an AAR can be built in the same way as the "full-megazord", and simply avoid depending on the rust
 crates for components that are not required.
 
-To help ensure this replacement is done safely at runtime, the `mozilla.appservices.support.native` provides
-helper functions for loading the correct megazord `.so` file.  The Kotlin wrapper for each component should
-load its shared library by calling `mozilla.appservices.support.native.loadIndirect`, specifying both the
-name of the component and the expected version number of the shared library.
+To help ensure this replacement is done safely at runtime, the `mozilla.appservices.support.native` package
```
Comment (rfk) [inline docs/design/megazords.md:55]: All I did here was add the word "package" that appears at the end of this line, but then reformatted the rest of the paragraph to maintain the pleasing visual indent.

**Comment by rfk** [comment]
Asking everyone for review because I want to give every opportunity for folks to talk me down from this opinionated position :-)

**Review by mhammond — APPROVED** [review]
👍

**Review by eoger — APPROVED** [review]
kill it!

**Comment by rfk** [comment]
@lougeniaC64 give me an r+ for this in slack, I'm going to go ahead and merge.

**Pull request merged**